internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,661 @@
+				var interopSettingsValue interface{}
+				err := json.Unmarshal([]byte(plan.InteropSettings), &interopSettingsValue)
+				if err != nil {
+					return err
+				}
+
+				model.Properties.InteropSettings = &interopSettingsValue
+			}
+
+			if metadata.ResourceData.HasChange("local_diagnostics_access") {
+				model.Properties.LocalDiagnosticsAccess = expandPacketCoreControlLocalDiagnosticsAccessConfiguration(plan.LocalDiagnosticsAccess)
+			}
+
+			if metadata.ResourceData.HasChange("mobile_network_id") {
+				model.Properties.Sites = expandPacketCoreControlPlaneSites(plan.SiteIds)
+			}
+
+			if metadata.ResourceData.HasChange("platform") {
+				model.Properties.Platform = expandPlatformConfigurationModel(plan.Platform)
+			}
+
+			if metadata.ResourceData.HasChange("sku") {
+				model.Properties.Sku = packetcorecontrolplane.BillingSku(plan.Sku)
+			}
+
+			if metadata.ResourceData.HasChange("version") && plan.SoftwareVersion != "" {
+				model.Properties.Version = &plan.SoftwareVersion
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				model.Tags = &plan.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r PacketCoreControlPlaneResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.PacketCoreControlPlaneClient
+
+			id, err := packetcorecontrolplane.ParsePacketCoreControlPlaneID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := PacketCoreControlPlaneModel{
+				Name:              id.PacketCoreControlPlaneName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+
+				state.Identity, err = flattenMobileNetworkUserAssignedToNetworkLegacyIdentity(model.Identity)
+				if err != nil {
+					return fmt.Errorf("flattening `identity`: %+v", err)
+				}
+
+				properties := model.Properties
+
+				state.UeMtu = pointer.From(model.Properties.UeMtu)
+
+				state.ControlPlaneAccessIPv4Address = pointer.From(properties.ControlPlaneAccessInterface.IPv4Address)
+
+				state.ControlPlaneAccessIPv4Gateway = pointer.From(properties.ControlPlaneAccessInterface.IPv4Gateway)
+
+				state.ControlPlaneAccessIPv4Subnet = pointer.From(properties.ControlPlaneAccessInterface.IPv4Subnet)
+
+				state.ControlPlaneAccessName = pointer.From(properties.ControlPlaneAccessInterface.Name)
+
+				if properties.CoreNetworkTechnology != nil { // it still needs a nil check because it needs to do type conversion
```

Comment:
For readability can we put the comment on the line above
```suggestion
				// it still needs a nil check because it needs to do type conversion
				if properties.CoreNetworkTechnology != nil { 
```